ls1intum/Artemis
# `Iris`: More consistent error handling on client side

### Checklist
#### General


#### Server
#### Client
#### Changes affecting Programming Exercises

### Description
Centralized Exception Management: This update streamlines the way exceptions are handled throughout the client, providing a more consistent approach.

Improved Error Display: The new error handling mechanism ensures a consistent and user-friendly experience when errors occur. Error messages are now displayed to the user in a clear and informative manner, enhancing overall usability, in both languages.

Resend Message Button: In scenarios where a student encounters a failure while sending a message due to an unreachable endpoint, this PR introduces a "resend message" button. This button allows the student to attempt resending the message without needing to repeat the entire process.

Shaking Animation: If the student is still unable to send the message even after pressing the resend button, a visually engaging shaking animation appears. This animation effectively communicates the failure and prompts the student to take further action.

### Steps for Testing

Prerequisites:
- 1 Instructor
- 1 Student
- 1 Programming Exercise with Iris enabled

1. Log in to Artemis
2. Navigate to Course Administration
3. Open a programming exercise with Iris enabled.
4. Ensure that the endpoint used to send messages is not reachable.
5. Send a message using Iris.
6. Observe the behavior of the error handling mechanism when the message fails to send(both EN/DE).
7. Press the newly introduced "resend message" button.
8. Observe the shaking animation if the resend operation fails.
9. Make the endpoint reachable again.
10. Press the retry button.
11. Observe the successful disappearance of the error box, indicating that the message has been sent successfully.

### Review Progress






#### Performance Review
#### Code Review
#### Manual Tests
#### Exam Mode Test

### Test Coverage







### Screenshots
